crates/pallet-bridges-initializer-currency-swap/src/lib.rs

```diff
@@ -0,0 +1,397 @@
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+            upgrade_init::pre_upgrade::<T>()
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade(state: Vec<u8>) -> Result<(), &'static str> {
+            upgrade_init::post_upgrade::<T>(state)
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Verify if currencies are balanced.
+        #[pallet::call_index(0)]
+        #[pallet::weight(T::WeightInfo::verify_balanced())]
+        pub fn verify_balanced(_origin: OriginFor<T>) -> DispatchResult {
+            if !Pallet::<T>::is_balanced()? {
+                return Err(Error::<T>::NotBalanced.into());
+            }
+
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    /// Initialize bridges pot accounts.
+    pub fn initialize() -> Result<Weight, DispatchError> {
+        let mut weight = T::DbWeight::get().reads(0);
+
+        with_storage_layer(move || {
+            let evm_total_issuance = T::EvmCurrency::total_issuance();
+            let evm_bridge_balance = T::EvmCurrency::total_balance(&T::EvmNativeBridgePot::get());
+            weight += T::DbWeight::get().reads(2);
+
+            let evm_swappable = evm_total_issuance
+                .checked_sub(&evm_bridge_balance)
+                .expect("evm_total_issuance is greater than evm_bridge_balance; qed.");
+
+            let native_swap_reserved = T::BalanceConverterEvmToNative::convert(evm_swappable);
+            let native_bridge_balance = native_swap_reserved
+                .checked_add(&T::NativeCurrency::minimum_balance())
+                .ok_or(ArithmeticError::Overflow)?;
+            weight += T::DbWeight::get().reads(1);
+
+            weight += Self::make_native_bridge_balance_be(native_bridge_balance)?;
+
+            let native_total_issuance = T::NativeCurrency::total_issuance();
+            weight += T::DbWeight::get().reads(1);
+
+            let native_swappable = native_total_issuance
+                .checked_sub(&native_bridge_balance)
+                .expect("native_total_issuance is greater than native_bridge_balance; qed.");
+
+            let evm_swap_reserved = T::BalanceConverterNativeToEvm::convert(native_swappable);
+            let evm_bridge_balance = evm_swap_reserved
+                .checked_add(&T::EvmCurrency::minimum_balance())
+                .ok_or(ArithmeticError::Overflow)?;
+            weight += T::DbWeight::get().reads(1);
+
+            weight += Self::make_evm_bridge_balance_be(evm_bridge_balance)?;
+
+            if !Self::is_balanced()? {
+                return Err::<(), DispatchError>(Error::<T>::NotBalanced.into());
+            }
+            weight += T::DbWeight::get().reads(8);
+
+            Ok(())
+        })?;
+
+        Ok(weight)
+    }
+
+    /// Make native bridge balance be provided amount value.
+    ///
+    /// The logic can change native swappable balance value.
```

Comment:
```suggestion
    /// Make native bridge balance be provided amount value.
    ///
    /// This function TRANSFERS the tokens to/from the treasury to balance the bridge pots.
    /// It will not change the total issuance, but it can change the native swappable balance value.
```

Reply:
How about we add `debug_assert!`/test to this fn to make sure we don't alter the total issuance?

Reply:
Thanks, agree - https://github.com/humanode-network/humanode/pull/716/commits/7bacc72c823577f6d100185ad4c0433f25d64edf.